.github/workflows/e2e.yml

```diff
@@ -0,0 +1,298 @@
+
+      - name: Start minikube
+        run: |
+          curl -LO \
+            https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64 \
+            && install minikube-linux-amd64 /tmp/
+  
+            kv=$(curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt)
+            curl -LO \
+            https://storage.googleapis.com/kubernetes-release/release/$kv/bin/linux/amd64/kubectl \
+            && install kubectl /tmp/
+  
+            /tmp/minikube-linux-amd64 config set WantUpdateNotification false
+            /tmp/minikube-linux-amd64 start --driver=docker
+
+
+      - name: Configure Docker Socket
+        run: eval $(minikube docker-env)
+
+      - name: Load image
+        run: |
+          docker load --input /tmp/api-server/api-server.tar
+          docker load --input /tmp/coordinator/coordinator.tar
+          docker load --input /tmp/dashboard/dashboard.tar
+          docker load --input /tmp/metadata-dashboard/metadata-dashboard.tar
+          docker load --input /tmp/metadata/metadata.tar
+          docker load --input /tmp/serving/serving.tar
+          docker load --input /tmp/worker/worker.tar
+          minikube image load local/api-server:stable
+          minikube image load local/coordinator:stable
+          minikube image load local/dashboard:stable
+          minikube image load local/metadata-dashboard:stable
+          minikube image load local/metadata:stable
+          minikube image load local/serving:stable
+          minikube image load local/worker:stable
+          minikube image ls
+
+      - name: Helm Install
+        run: |
+          helm repo add jetstack https://charts.jetstack.io
+          helm repo update
+          helm install certmgr jetstack/cert-manager --set installCRDs=true --version v1.8.0 --namespace cert-manager --create-namespace
+          helm install featureform ./charts/featureform --set global.repo=local --set global.pullPolicy=Never
+          helm install quickstart ./charts/quickstart
+
+      - name: Check Pods
+        run: |
+          seconds=0
+          while [ $(kubectl get pods | grep "featureform" | grep -wv featureform-quickstart | awk {'print $3'} | column -t | grep -wv Running -c ) != 0 ]
+          do 
+            echo "Waiting for containers to be ready: $(kubectl get pods | grep "featureform" | grep -wv featureform-quickstart | awk {'print $3'} | column -t | grep -wv Running -c ) Remaining"
+            seconds=$((seconds + 10))
+            sleep 10
+            if (( seconds > 360 ))
+            then
+            	kubectl get pods
+            	echo "**Containers Did Not Ready**"
+            	exit 1
+            fi
+          done
+          echo "Containers Ready"
+          kubectl get pods
+        shell: bash
+
+      - name: Enable Minikube Ingress
+        run: |
+          kubectl port-forward svc/featureform-ingress-nginx-controller 8000:443 7000:80 &
+          sleep 2
```

Comment:
is this always enough time to guarantee port forwarding is up and running. alternative would be to run a curl do while loop until it returns true to guarantee next command doesn't run until it works. Very minor though

do curl:7000
timer=0
while curl is false and timer <10
  timer++
  curl:7000
  sleep 1

Reply:
Changed this as well